File: src/sequence/menuitem.cpp
#include "menuitem.h"

namespace ks {

MenuItem::MenuItem(void(*callback)(ks::SceneManager& scene))
    : _callback(callback) {}

SequenceType MenuItem::type() const {
    return ks::SequenceType::Menu;
}

SequenceItem* MenuItem::clone() const {
    return new MenuItem(*this);
}

void MenuItem::run(ks::SceneManager& scene) {
    if (_callback) {
        _callback(scene);
    }
}

bool MenuItem::finished() const {
    return true;
}

} // namespace ks

File: tools/converter/src/dto/menu_item.py
from src.dto.sequence_item import SequenceItem, SequenceType


class MenuItem(SequenceItem):
    def some_method(self):
        pass

    def __init__(self, function_callback: str):
        super().__init__(SequenceType.MENU)
        self.function_callback = function_callback

    def __str__(self):
        return f"MenuItem(function_callback={self.function_callback})"
